Update Log proxy instruction (#7444)
DataDog/documentation: content/en/agent/logs/proxy.md

Log collection requires the Datadog Agent v6.0+. Older versions of the Agent do not include the `log collection` interface.

> Since agent version v6.14/v7.14 Datadog recommend to use and enforce HTTPS transport (see [agent Transport for Logs][2]). 

Reviewer: ```suggestion
As of Agent v6.14/v7.14, Datadog recommends the use of and enforcing HTTPS transport (see [agent Transport for Logs][2]). 
```

> As of Agent v6.14/v7.14, Datadog recommends the use of and enforcing HTTPS transport (see [agent Transport for Logs][2]). 

Reviewer 2: I would recommend to have some **Bold** around HTTPS to insist on it.

Author: done

> If you are using the HTTPS transport for logs, please reffer to the [agent proxy documentation][1] and use the same set of proxy settings as other data types.

Reviewer: ```suggestion
If you are using the HTTPS transport for logs, please refer to the [agent proxy documentation][1] and use the same set of proxy settings as other data types.
```

* `DD_LOGS_CONFIG_SOCKS5_PROXY_ADDRESS`

{{% /tab %}}
{{< /tabs >}}

## Examples of TCP proxy

### Using HAProxy as a TCP Proxy for Logs

> 

Reviewer 2: So I would actually start with the Agent configuration.

As it is a shorter part, it might be more impactful to have it first, what do you think?

Reviewer: +1 to this idea, moving up the agent config section will help for engagement

Author: done

# For HAProxy 1.8 and newer
resolvers my-dns
    nameserver dns1 <DNS_SERVER_IP>:53
    nameserver dns2 <DNS_SECONDARY_SERVER_IP>:53
    resolve_retries 3
    timeout resolve 2s
    timeout retry 1s
    accepted_payload_size 8192
    hold valid 10s
    hold obsolete 60s
    
# This declares the endpoint where your Agents connects for
# sending Logs (e.g the value of "logs.config.logs_dd_url")
frontend logs_frontend
    bind *:10514
    mode tcp
    option tcplog
    default_backend datadog-logs
    
# This is the Datadog server. In effect any TCP request coming
# to the forwarder frontends defined above are proxied to
# Datadog's public endpoints.
backend datadog-logs
    balance roundrobin
    mode tcp
    option tcplog
    server datadog agent-intake.logs.datadoghq.com:10516 ssl verify required ca-file /etc/ssl/certs/ca-certificates.crt check port 10516
```

**Note**: Download the certificate with the following command:
        * `sudo apt-get install ca-certificates` (Debian, Ubuntu)
        * `yum install ca-certificates` (CentOS, Redhat)
If successful, the file will be located at `/etc/ssl/certs/ca-bundle.crt` for CentOS, Redhat.

> Once the HAProxy configuration is in place, you can reload it or restart HAProxy. **It is recommended to have a `cron` job that reloads HAProxy every 10 minutes** (usually doing something like `service haproxy reload`) to force a refresh of HAProxy's DNS cache, in case `app.datadoghq.com` fails over to another IP.

Reviewer: ```suggestion
Once the HAProxy configuration is in place, you can reload it or restart HAProxy. **It is recommended to have a `cron` job that reloads HAProxy every 10 minutes** (for example, `service haproxy reload`) to force a refresh of HAProxy's DNS cache, in case `app.datadoghq.com` fails over to another IP.
```

    retries 3
    option  redispatch
    timeout client 5s
    timeout server 5s
    timeout connect 5s

# This declares a view into HAProxy statistics, on port 3833
# You do not need credentials to view this page and you can
# turn it off once you are done with setup.
listen stats
    bind *:3833
    mode http
    stats enable
    stats uri /

# This section is to reload DNS Records
# Replace <DNS_SERVER_IP> and <DNS_SECONDARY_SERVER_IP> with your DNS Server IP addresses.
# For HAProxy 1.8 and newer
resolvers my-dns
    nameserver dns1 <DNS_SERVER_IP>:53
    nameserver dns2 <DNS_SECONDARY_SERVER_IP>:53
    resolve_retries 3
    timeout resolve 2s
    timeout retry 1s
    accepted_payload_size 8192
    hold valid 10s
    hold obsolete 60s
    
# This declares the endpoint where your Agents connects for
# sending Logs (e.g the value of "logs.config.logs_dd_url")
frontend logs_frontend
    bind *:10514
    mode tcp
    default_backend datadog-logs
    
# This is the Datadog server. In effect any TCP request coming
# to the forwarder frontends defined above are proxied to
# Datadog's public endpoints.
backend datadog-logs
    balance roundrobin
    mode tcp
    option tcplog

>     server datadog agent-intake.logs.datadoghq.eu:443 ssl verify required ca-file /etc/ssl/certs/ca-bundle.crt check port 443

Reviewer 2: This is the EU endpoint, how do we make sure that for US we have the right endpoint?

Reviewer 2: See in the nginx example:

```
{{< /site-region >}}
{{< site-region region="eu" >}}
```

Reviewer: Yep, correct example! It's `{{< site-region region="us" >}}` for US content, `{{< site-region region="eu" >}}` for EU content.

Author: sorry I don't get understand the comment. is there any problem with the current version? 
 US example is from line 82 to 149 and EU example is from 150 to 216  for HAProxy. Should I change something?

Reviewer 2: Seems that both EU and US version points to the `.eu` endpoint currently.

So we would need to include the `site-region` header to make sure we display the right endpoint based on the region.

Reviewer: If you format it like:

{{< site-region region="us" >}}
US content goes here
{{< /site-region >}}

{{< site-region region="us" >}}
EU content goes here
{{< /site-region >}}

It will switch the content based on region when someone uses the region selector in the top right of the docs page.

Author: double checked on slack with sarina, we are good :D

```

**Note**: Download the certificate with the following command:

* `sudo apt-get install ca-certificates` (Debian, Ubuntu)
* `yum install ca-certificates` (CentOS, Redhat)

If successful, the file will be located at `/etc/ssl/certs/ca-bundle.crt` for CentOS, Redhat.

> Once the HAProxy configuration is in place, you can reload it or restart HAProxy. **It is recommended to have a `cron` job that reloads HAProxy every 10 minutes** (usually doing something like `service haproxy reload`) to force a refresh of HAProxy's DNS cache, in case `app.datadoghq.eu` fails over to another IP.

Reviewer: ```suggestion
Once the HAProxy configuration is in place, you can reload it or restart HAProxy. **It is recommended to have a `cron` job that reloads HAProxy every 10 minutes** (for example, `service haproxy reload`) to force a refresh of HAProxy's DNS cache, in case `app.datadoghq.eu` fails over to another IP.
```

{{< /site-region >}}

#### Agent configuration

> Then edit the `datadog.yaml` Agent configuration file and set `logs_no_ssl` to `true`. This is needed as HAProxy does not simply forward the traffic and is not the Datadog backend so cannot use the same certificate.

Reviewer: ```suggestion
Edit the `datadog.yaml` Agent configuration file and set `logs_no_ssl` to `true`. This is needed as HAProxy does not forward the traffic and is not the Datadog backend, so you cannot use the same certificate.
```

**Note**: `logs_no_ssl` might set to true because HAProxy is configured to encrypt the data. Do not set this parameter to `true` otherwise.

```
logs_config:
  use_tcp: true
  logs_dd_url: "<PROXY_SERVER_DOMAIN>:10514"
  logs_no_ssl: true
```

### Using NGINX as a TCP Proxy for logs

> 

Reviewer 2: Same here, we might want to have the Agent configuration first.

Author: done

{{< site-region region="us" >}}

```conf
user nginx;
worker_processes auto;
error_log /var/log/nginx/error.log;
pid /run/nginx.pid;
events {
    worker_connections 1024;
}
# TCP Proxy for Datadog Agent
stream {
    server {
        listen 10514; #listen for logs
        proxy_ssl on;
        proxy_pass agent-intake.logs.datadoghq.com:10516;
    }
}
```

{{< /site-region >}}
{{< site-region region="eu" >}}

```conf
user nginx;
worker_processes auto;
error_log /var/log/nginx/error.log;
pid /run/nginx.pid;
events {
    worker_connections 1024;
}
# TCP Proxy for Datadog Agent
stream {
    server {
        listen 10514; #listen for logs
        proxy_ssl on;
        proxy_pass agent-intake.logs.datadoghq.eu:443;
    }
}
```

{{< /site-region >}}

#### Agent configuration

Edit the `datadog.yaml` Agent configuration file and set `logs_config.logs_dd_url` to use the newly created proxy instead of establishing a connection directly with Datadog:

```yaml
logs_config:
  use_tcp: true
  logs_dd_url: myProxyServer.myDomain:10514
```

> **Note**: Do not change the `logs_no_ssl` parameter as NGINX is simply forwarding the traffic to Datadog and does not decrypt or encrypt the traffic.

Reviewer: ```suggestion
**Note**: Do not change the `logs_no_ssl` parameter as NGINX is forwarding the traffic to Datadog and does not decrypt or encrypt the traffic.
```
